modules/ip6/datapath/ndp_ns_output.c

```diff
@@ -69,6 +70,11 @@ static uint16_t ndp_ns_output_process(
 			next = ERROR;
 			goto next;
 		}
+		if (nh->type == GR_NH_T_BLACKHOLE || nh->type == GR_NH_T_REJECT) {
+			next = BLACKHOLE;
+			goto next;
```

Comment:
You didn't use the same output node for GR_NH_T_REJECT here (icmp6_output_admin_prohibited?)